Web SDK - Content-Security-Policy (#2183)
braze-inc/braze-docs: _docs/_developer_guide/platform_integration_guides/web/csp_headers.md

---
nav_title: Content Security Policy
platform: Web
page_order: 50
page_type: reference
description: "This article covers Content-Security-Policy headers needed with the Braze Web SDK"
---

# Content Security Policy Headers

Content-Security-Policy provides added security by restricting how and where content can be loaded on your website.

{% alert important %}
This article is intended for developers working on websites that enforce CSP rules, and how to integrate with Braze. It is not intended as advice on how you should approach security.
{% endalert %}

## Nonce Attributes

If you use a `nonce` value in your `script-src` or `style-src` directives, pass that value to the `contentSecurityNonce` initialization option to propagate it to newly created scripts and styles generated by the SDK.

```javascript
import braze from "@braze/web-sdk";

braze.initialize(apiKey, {
  baseUrl: baseUrl,

>   contentSecurityNonce: "YWJjMTIz", // assumes a "nonce-YWJjMTIz" CSP value

Reviewer: might be better to replace this with `"YOUR-NONCE-HERE"` or something so everybody doesn't literally use `YWJjMTIz`

>   contentSecurityNonce: "YWJjMTIz", // assumes a "nonce-YWJjMTIz" CSP value

Author: ```suggestion
  contentSecurityNonce: "YOUR-NONCE-HERE", // assumes a "nonce-YOUR-NONCE-HERE" CSP value
```

});
```

## Directives

### connect-src

- `connect-src https://sdk.iad-01.braze.com` - allows the SDK to communicate with Braze APIs.
  - Change this URL to match your `baseUrl` initialization option's API Endpoint, as found [here](https://www.braze.com/docs/user_guide/administrative/access_braze/sdk_endpoints/).

### script-src

- `script-src https://js.appboycdn.com` - required when using the CDN-hosted integration.
- `script-src 'unsafe-eval'` - required only when using the integration snippet which contains reference to `appboyQueue`

>   - To avoid using this directive, integrate using NPM or reference the `appboy.min.js` library directly via `<script src="path/to/appboy.min.js"></script>` without the `appboyQueue` logic.

Reviewer 2: This isn't true because the non-`npm` versions will still have `eval`. So you'd have to download through `npm` and then reference it like this or self-host it.

Also, we should include nonces here since they're required for HTML IAMs that have `<script>` blocks (which I assume most do).

>   - To avoid using this directive, integrate using NPM or reference the `appboy.min.js` library directly via `<script src="path/to/appboy.min.js"></script>` without the `appboyQueue` logic.

Author: ```suggestion
  - To avoid using this directive, integrate using NPM instead.
```

## Font Awesome

> If you use Font Awesome icons within your dashboard campaigns, the following CSP directives are required:

Reviewer: `If you use Font Awesome icons within your dashboard campaigns` does this actually matter? I don't think it does? I think this may be necessary whether or not their campaigns actually use font awesome?  

Or maybe you're just saying that if they don't use them, they can just disable fontawesome as described below in "Alternatively"? If that's the case, I might reverse the order, since FA usage is on its way out anyway, so it's like, if you don't use fontawesome, use doNotLoadFontAwesome, alternatively, add the following CSP directives.

- `style-src https://use.fontawesome.com`

> - `style-src 'unsafe-inline'` or `style-src 'nonce-...'` are also required.

Reviewer 2: Maybe put the nonce option first?

> - `style-src 'unsafe-inline'` or `style-src 'nonce-...'` are also required.

Author: ```suggestion
- `style-src 'nonce-...'` or `style-src 'unsafe-inline'` are also required.
```
